### ISSUE TITLE
Unable to download audio with `filesize_approx` filter

### ISSUE THREAD

**Issue by arbitrary-dev** [yt-dlp/yt-dlp#8730]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

Unable to download audio with `filesize` filter given only estimate of the actual size available.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ yt-dlp -vU
[debug] Command-line config: ['-vU']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.11.16 from yt-dlp/yt-dlp [24f827875]
[debug] Python 3.11.5 (CPython x86_64 64bit) - Linux-6.5.6-gentoo-x86_64-AMD_Ryzen_9_5900HX_with_Radeon_Graphics-with-glibc2.37 (OpenSSL 3.0.11 19 Sep 2023, glibc 2.37)
[debug] exe versions: ffmpeg 6.0 (fdk,setts), ffprobe 6.0
[debug] Optional libraries: certifi-3021.03.16, mutagen-1.47.0, pycrypto-3.19.0, requests-2.31.0, sqlite3-3.43.2, urllib3-2.1.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests
[debug] Loaded 1901 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2023.11.16 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2023.11.16 from yt-dlp/yt-dlp)

$ yt-dlp --format="bestaudio[filesize<=14M]" https://melodium.bandcamp.com/track/in-deserto-2
[Bandcamp] Extracting URL: https://melodium.bandcamp.com/track/in-deserto-2
[Bandcamp] in-deserto-2: Downloading webpage
ERROR: [Bandcamp] 2112470957: Requested format is not available. Use --list-formats for a list of available formats

$ yt-dlp --list-formats https://melodium.bandcamp.com/track/in-deserto-2
[Bandcamp] Extracting URL: https://melodium.bandcamp.com/track/in-deserto-2
[Bandcamp] in-deserto-2: Downloading webpage
[info] Available formats for 2112470957:
ID      EXT RESOLUTION │ FILESIZE  TBR PROTO │ VCODEC     ACODEC  ABR
─────────────────────────────────────────────────────────────────────
mp3-128 mp3 audio only │ ~3.53MiB 128k https │ audio only mp3    128k
```

**Comment by pukkandan** [comment]
`filesize` is only the exact filesize. Use `filesize_approx` for the approximate one.

**Comment by arbitrary-dev** [comment]
```
$ yt-dlp --list-formats https://melodium.bandcamp.com/track/in-deserto-2
[Bandcamp] Extracting URL: https://melodium.bandcamp.com/track/in-deserto-2
[Bandcamp] in-deserto-2: Downloading webpage
[info] Available formats for 2112470957:
ID      EXT RESOLUTION │ FILESIZE  TBR PROTO │ VCODEC     ACODEC  ABR
─────────────────────────────────────────────────────────────────────
mp3-128 mp3 audio only │ ~3.53MiB 128k https │ audio only mp3    128k

$ yt-dlp --format="bestaudio[filesize_approx<=14M]" https://melodium.bandcamp.com/track/in-deserto-2
[Bandcamp] Extracting URL: https://melodium.bandcamp.com/track/in-deserto-2
[Bandcamp] in-deserto-2: Downloading webpage
ERROR: [Bandcamp] 2112470957: Requested format is not available. Use --list-formats for a list of available formats
```

**Comment by bashonly** [comment]
Regression in 39f32f1715c0dffb7626dda7307db6388bb7abaa
